Code/src/main/java/nl/utwente/group10/ui/components/lines/Connection.java

```diff
@@ -138,162 +162,138 @@ public boolean tryAddAnchor(ConnectionAnchor anchor, boolean overrideExisting, b
      */
-    public final boolean isConnected() {
-        return startAnchor.isPresent() && endAnchor.isPresent();
+    public boolean tryAddAnchor(ConnectionAnchor anchor) {
+        return tryAddAnchor(anchor,
+                ConnectionCreationManager.CONNECTIONS_OVERRIDE_EXISTING);
     }
 
     /**
-     * Removes this Connection, disconnecting its anchors and removing this Connection from the pane it is on.
+     * Tries to add an unspecified ConnectionAnchor to the connection.
+     *
+     * @param anchor
+     *            Anchor to add
+     * @param overrideExisting
+     *            If set, old anchors could be disconnected to make room for
+     *            this new one.
+     * @return Whether or not the anchor was added.
      */
-    public final void remove() {
-        disconnect();
-        getPane().getChildren().remove(this);
+    public boolean tryAddAnchor(ConnectionAnchor anchor, boolean overrideExisting) {
+        Optional<? extends ConnectionAnchor> slot = getAnchorSlot(anchor);
+
+        if (!slot.isPresent() || overrideExisting) {
+            setAnchor(anchor);
+            return true;
+        }
+
+        return false;
     }
 
     /**
      * Properly disconnects the given anchor from this Connection, notifying the anchor of its disconnection.
      */
     public final void disconnect(ConnectionAnchor anchor) {
+        boolean disconnected = false;
+        boolean wasConnected = isConnected();
+        // Find out what anchor to disconnect, and do so.
         if (startAnchor.isPresent() && startAnchor.get().equals(anchor)) {
             startAnchor = Optional.empty();
-            anchor.disconnectConnection(this);
-            ConnectionCreationManager.nextConnectionState();
-        }
-        if (endAnchor.isPresent() && endAnchor.get().equals(anchor)) {
+            disconnected = true;
+        } else if (endAnchor.isPresent() && endAnchor.get().equals(anchor)) {
             endAnchor = Optional.empty();
+            disconnected = true;
+        }
+        
+        if(disconnected) {
+            // Fully disconnect the anchor from this Connection.
+            removeListeners(anchor);
             anchor.disconnectConnection(this);
-            ConnectionCreationManager.nextConnectionState();
+            
+            if (wasConnected) {
+                ConnectionCreationManager.nextConnectionState();
+    
+                //Let the now disconnected anchor update its visuals.
+                anchor.getBlock().setConnectionState(ConnectionCreationManager.getConnectionState());
+                //Let the remaining connected anchors update their visuals.
+                startAnchor.ifPresent(a -> a.getBlock().setConnectionState(ConnectionCreationManager.getConnectionState()));
+                endAnchor.ifPresent(a -> a.getBlock().setConnectionState(ConnectionCreationManager.getConnectionState()));
```

Comment:
Could refactor out `ConnectionCreationManager.getConnectionState()`, mostly because of its very long name